packages/server/lib/browsers/firefox.ts

```diff
@@ -458,7 +458,10 @@ export async function open (browser: Browser, url: string, options: BrowserLaunc
   const [
     foxdriverPort,
     marionettePort,
-  ] = await Promise.all([getPort(), getPort()])
+  ] = await Promise.all([
+    Number(process.env.CYPRESS_FOXDRIVER_PORT) || getPort(),
+    Number(process.env.CYPRESS_MARIONETTE_PORT) || getPort(),
+  ])
```

Comment:
> I think this could also be done for the port that is currently covered by the environment variable CYPRESS_REMOTE_DEBUGGING_PORT and which is passed to Firefox via the command line option --remote-debugging-port <port>.
When I run Firefox with --remote-debugging-port 0 it binds to a random port (in the example output below this happens to be 32819) and outputs it as part of the following lines:
>
> WebDriver BiDi listening on ws://127.0.0.1:32819
DevTools listening on ws://127.0.0.1:32819/devtools/browser/69d2d46c-4d87-4505-b51b-cf53fa299317

This sounds reasonable to me to allow this port to be dynamically set for the remote-bugging-port as well. We would want do do something like the following to continue to honor `CYPRESS_REMOTE_DEBUGGING_PORT`.
```
remotePort =  Number(process.env.CYPRESS_REMOTE_DEBUGGING_PORT) || '0'
```

Also we would need to confirm this works/the output is the same with Firefox 86... That is or minimum supported Firefox version.

Reply:
Though are you pulling this from `stderr`? that is where I am seeing this which is odd

Reply:
I was looking at both streams combined and didn't realize the messages were output to distinct streams but now that I am looking for it that does seem to be the case. It does feel a little inconsistent but I also see this in Firefox code:

```Cu.printStderr(`WebDriver BiDi listening on ${this.address}\n`);```
and
```Cu.printStderr(`DevTools listening on ${this.address}\n`);```
but for Marionette
```lazy.logger.info(`Listening on port ${this.port}`);```
which with `browser.dom.window.dump.enabled` apparently outputs to stdout.

I tried looking at Firefox 86 code but I am not that familiar with Mercurial.
What I think I see is that `WebDriver BiDi listening on ` does not exist yet but the `DevTools listening on` message which also contains the same port could work. The Marionette port message looks like it could work (same string / message but different output code).

Reply:
Interesting.. Wonder why it's printing to stderr?? Maybe because it's meant for debugging and not "production" browser launches 😅

Reply:
FWIW I did find it (the fact that it outputs to stderr, the message and I suppose intended use) documented here:
https://firefox-source-docs.mozilla.org/remote/cdp/Usage.html
Also seem to say that this happens to be implemented since Firefox 86 exactly :smiley: